### DISCUSSION TITLE
Middleware request header always about:client

### DISCUSSION THREAD

**Discussion by kristofferso** [vercel/next.js#36255] [Help]
Hey everyone!
I've just implemented a simple analytics system by using the new middleware-functionality in Next.js 12. I want to store pageviews and their referrer (among other data), but this value is always _about:client_, meaning that there is no referrer data being passed through. I have tested this in production by linking to my site from another, and in the browser I can access the referrer in `document.referrer`. This however, does not propagate to the middleware function through the NextRequest. Other parameters in the request seem to be working, except from the `geo` prop.

The site is hosted on Netlify.

Any ideas here? Thanks 🙏

**Comment by dubbajubba** [comment]
@kristofferso, were you able to solve this? I am also having the same problem. The referrer is always `about: client`, `undefined`, or the page that I'm viewing, but never the referring page.

**Reply by kristofferso** [reply]
No, unfortunately I didn't. I suspect this works differently if you deploy it to Vercel, than Netlify. That was my next move, but I've been working on other stuff in the mean time. Let me know if you try it out!

(sorry about the late reply)
